resource_pool/pool_client/resource_pool_client.py

```diff
@@ -0,0 +1,103 @@
+from contextlib import asynccontextmanager
+from datetime import timedelta
+from typing import AsyncGenerator, Optional
+
+from temporalio import workflow
+
+from resource_pool.resource_pool_workflow import ResourcePoolWorkflow
+from resource_pool.shared import (
+    AcquiredResource,
+    AcquireRequest,
+    AcquireResponse,
+    DetachedResource,
+)
+
+
+# Use this class in workflow code that that needs to run on locked resources.
+class ResourcePoolClient:
+    def __init__(self, pool_workflow_id: str) -> None:
+        self.pool_workflow_id = pool_workflow_id
+        self.acquired_resources: list[AcquiredResource] = []
+        self.register_signal_handler()
+
+    def register_signal_handler(self) -> None:
+        signal_name = f"assign_resource_{self.pool_workflow_id}"
+        if workflow.get_signal_handler(signal_name) is None:
+            workflow.set_signal_handler(signal_name, self.assign_resource)
+        else:
+            raise RuntimeError(
+                f"{signal_name} already registered - if you use multiple ResourcePoolClients within the "
+                f"same workflow, they must use different pool_workflow_ids"
+            )
+
+    async def send_acquire_signal(self) -> None:
```

Comment:
Should prefix all methods in this class that you don't expect an end user to call with `_`